cmd/scale.go

```diff
@@ -389,10 +390,13 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 					}
 				}
 
-				osPublisher := vmss.VirtualMachineProfile.StorageProfile.ImageReference.Publisher
-				if osPublisher != nil && (strings.EqualFold(*osPublisher, "MicrosoftWindowsServer") || strings.EqualFold(*osPublisher, "microsoft-aks")) {
-					_, _, winPoolIndex, _, err = utils.WindowsVMNameParts(vmName)
-					log.Errorln(err)
+				if sc.agentPool.OSType == api.Windows {
+					winPoolIndexStr := vmssName[len(vmssName)-2:]
```

Comment:
This is bad. See:

https://github.com/Azure/aks-engine/blob/master/pkg/api/types.go#L916

Essentially, we bake a two digit integer into the pool name, which we are only able to determine conclusively at cluster creation time by enumerating through the api model array. There's nothing JSON-breaking that would prevent the api model from being changed over time in such a way where the `agentPoolProfiles` array items get re-ordered. So the only way we can determine the index that cluster creation appended into the VMSS name is to get the last two chars and convert it to an int.

And then we use that int starting in Line 466 below.

I think this solution is the most practical way forward without a significant refactor.